### PR TITLE
Implemented Distance type

### LINKED ISSUE #21: Create NOX.Type "Distance"
MP-6932

### PR DESCRIPTION
**Pull request by dejobratic**
- Created a Distance value object with QuantityValue as underlying type.
- Distance can be created with kilometers and miles as units from a numeric value or two LatLong objects that represent the origin and destination. In the latter case the distance is calculated using the Haversine formula http://www.movable-type.co.uk/scripts/latlong.html. Additional implementations can be added as well.
- ToString method prints the value and symbol for unit.

### CODE REVIEW

File: src/Nox.Types.EntityFramework/DistanceConverter.cs
Comment (rochar) [inline src/Nox.Types.EntityFramework/DistanceConverter.cs:1]: Can we move this class to a subFolder called ./Types/Distance? keeping the namespace, and having the same folder struct in Nox.Types,NoxTypes.EntityFramework, and Nox.Types.Tests?

File: src/Nox.Types.EntityFramework/DistanceConverter.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using Microsoft.EntityFrameworkCore.Storage.ValueConversion;
+
+namespace Nox.Types.EntityFramework;
+
+public class DistanceToKilometerConverter : ValueConverter<Distance, double>
+{
+    public DistanceToKilometerConverter() : base(area => (double)area.ToKilometers(), d => Distance.FromKilometers(d)) { }
```
Comment (rochar) [inline src/Nox.Types.EntityFramework/DistanceConverter.cs]: area variable should be called distance instead?

File: src/Nox.Types.EntityFramework/DistanceConverter.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+}
+public class DistanceToMilesConverter : ValueConverter<Distance, double>
+{
+    public DistanceToMilesConverter() : base(area => (double)area.ToMiles(), d => Distance.FromMiles(d)) { }
```
Comment (rochar) [inline src/Nox.Types.EntityFramework/DistanceConverter.cs]: a suggestion, call d distanceValue for example . So its easier to read?

File: tests/Nox.Types.Tests/NoxDistanceTests.cs
Comment (rochar) [inline tests/Nox.Types.Tests/NoxDistanceTests.cs:1]: Can we move this class to a subFolder called ./Types/Distance?
And rename the class to be DistanceTests
We are working on setting this conventions, maybe you can start following what will be propose, thks a lot

**Review by rochar — CHANGES_REQUESTED** [review]
Looks good, just left a few suggestion
Thks

File: tests/Nox.Types.Tests/NoxDistanceTests.cs
Comment (dejobratic) [reply tests/Nox.Types.Tests/NoxDistanceTests.cs:1]: Thanks for the suggestions. All test classes and EF converters have been renamed and moved to dedicated subfolder (.Types/<type-name>).

**Review by rochar — DISMISSED** [review]

**Comment by rochar** [comment]
@dejobratic on the commit message you can relate to your issue, adding to the description #<issuenumber>

On the issue you can also select your branch or PR:
image

**Review by jan-schutte — APPROVED** [review]
Nicely done @dejobratic

**Pull request merged**